staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go

```diff
@@ -168,14 +184,359 @@ func TestSimpleCRUD(t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer noxuWatch1.Stop()
+
+	createdNoxuInstance1, err := instantiateCustomResource(t, testserver.NewNoxuInstance(ns1, "foo"), noxuResourceClient1, noxuDefinition)
+	if err != nil {
+		t.Fatalf("unable to create noxu Instance:%v", err)
+	}
+
+	select {
+	case watchEvent := <-noxuWatch1.ResultChan():
+		if e, a := watch.Added, watchEvent.Type; e != a {
+			t.Errorf("expected %v, got %v", e, a)
+			break
+		}
+		createdObjectMeta1, err := meta.Accessor(watchEvent.Object)
+		if err != nil {
+			t.Fatal(err)
+		}
+		// it should have a UUID
+		if len(createdObjectMeta1.GetUID()) == 0 {
+			t.Errorf("missing uuid: %#v", watchEvent.Object)
+		}
+		if e, a := ns1, createdObjectMeta1.GetNamespace(); e != a {
+			t.Errorf("expected %v, got %v", e, a)
+		}
+		createdTypeMeta1, err := meta.TypeAccessor(watchEvent.Object)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if e, a := noxuDefinition.Spec.Group+"/"+noxuDefinition.Spec.Version, createdTypeMeta1.GetAPIVersion(); e != a {
+			t.Errorf("expected %v, got %v", e, a)
+		}
+		if e, a := noxuDefinition.Spec.Names.Kind, createdTypeMeta1.GetKind(); e != a {
+			t.Errorf("expected %v, got %v", e, a)
+		}
+
+	case <-time.After(5 * time.Second):
+		t.Errorf("missing watch event")
+	}
+
+	gottenNoxuInstance1, err := noxuResourceClient1.Get("foo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if e, a := createdNoxuInstance1, gottenNoxuInstance1; !reflect.DeepEqual(e, a) {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+
+	listWithItem1, err := noxuResourceClient1.List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if e, a := 1, len(listWithItem1.(*unstructured.UnstructuredList).Items); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+	if e, a := *createdNoxuInstance1, listWithItem1.(*unstructured.UnstructuredList).Items[0]; !reflect.DeepEqual(e, a) {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+
+	if err := noxuResourceClient1.Delete("foo", nil); err != nil {
+		t.Fatal(err)
+	}
+
+	listWithoutItem1, err := noxuResourceClient1.List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if e, a := 0, len(listWithoutItem1.(*unstructured.UnstructuredList).Items); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+
+	select {
+	case watchEvent := <-noxuWatch1.ResultChan():
+		if e, a := watch.Deleted, watchEvent.Type; e != a {
+			t.Errorf("expected %v, got %v", e, a)
+			break
+		}
+		deletedObjectMeta1, err := meta.Accessor(watchEvent.Object)
+		if err != nil {
+			t.Fatal(err)
+		}
+		// it should have a UUID
+		createdObjectMeta1, err := meta.Accessor(createdNoxuInstance1)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if e, a := createdObjectMeta1.GetUID(), deletedObjectMeta1.GetUID(); e != a {
+			t.Errorf("expected %v, got %v", e, a)
+		}
+
+	case <-time.After(5 * time.Second):
+		t.Errorf("missing watch event")
+	}
+
+	// Second namespace
+	ns2 := "namespace-2"
```

Comment:
Is this much different from namespace-1? Can we merge these two into a (local) func?